### ISSUE TITLE
Incorrect PYTHONPATH on Ubuntu 10.04

### ISSUE THREAD

**Issue by smits** [ros/catkin#635]
Hi all,

I'm trying to use catkin on an older Ubuntu version due to legacy reasons. I've installed python2.7 on it from ppa:fkrull/deadsnakes and made sure I have CMake 2.8.3. I'm using virtualenv to make sure that catkin uses python2.7 since all catkin shebangs are without python version. Although it seems catkin is building ok the generated _setup_util.py exports the wrong PYTHONPATH: 

```
grep PYTHONPATH /opt/ros/hydro/_setup_util.py

        'PYTHONPATH': 'lib/python./site-packages
```

For some reason the MAJOR and MINOR versions are empty. I've skimmed through the catkin sources but cannot find where this CMake variables get set.

**Comment by dirk-thomas** [comment]
The template used for the `_setup_util.py` file uses the CMake variable `PYTHON_INSTALL_DIR` (see https://github.com/ros/catkin/blob/9a5da56c7ecfc456c2cb73124432eaf2fcc482b2/cmake/templates/_setup_util.py.in#L58).

This CMake variable is computed in https://github.com/ros/catkin/blob/9a5da56c7ecfc456c2cb73124432eaf2fcc482b2/cmake/python.cmake#L53 but you should see a message during the configure step which states the Python version being used (https://github.com/ros/catkin/blob/9a5da56c7ecfc456c2cb73124432eaf2fcc482b2/cmake/python.cmake#L27).

There is a second code path for the tools catkin_make / catkin_make_isolated in Python. May be you can add some debug out there if the CMake part is not responsible: https://github.com/ros/catkin/blob/9a5da56c7ecfc456c2cb73124432eaf2fcc482b2/python/catkin/builder.py#L238

**Comment by wyc** [comment]
Please see #636

**Comment by smits** [comment]
The patch provided in #636 fixes my issue!

**Comment by dirk-thomas** [comment]
@smits Could you please provide some information after trying it on your machine why the Python version was not extracted for you? I think it would still be valuable to understand _why_ it failed in the first place - e.g. the proposed patch relies on invoking Python which is performance wise always something to reconsider.

**Comment by dirk-thomas** [comment]
@smits See previous comment.

**Comment by smits** [comment]
Update: I was using indigo-devel instead of groovy-devel. With the original groovy-devel my issue is resolved even without #636

**Comment by dirk-thomas** [comment]
Thank you for clarifying the reason why this happened for you.
